### DISCUSSION TITLE
Using custom serializers within Prefect Orion

### DISCUSSION THREAD

**Discussion by andersy005** [PrefectHQ/prefect#5152] [Q&A]
I am currently trying out Prefect Orion, and I'm enjoying the experience so far. I'd like to write a custom serializer for xarray objects and my serializer looks as follows:

```python
from prefect.orion.serializers import _SERIALIZERS, Serializer, register_serializer
import xarray as xr

@register_serializer("xarray")
class XarraySerializer(Serializer[xr.Dataset]):
    @staticmethod
    def dumps(data: xr.Dataset, path: str) -> bytes:
        xr.backends.api.to_zarr(path, data, mode='w')
        return path.encode() # To conform to the spec
    
    @staticmethod
    def loads(blob: bytes) -> xr.Dataset:
        path = blob.decode()
        ds = xr.open_dataset(path, engine="zarr")
        return ds
 ```

To use a custom serializer In 1st Gen Prefect is straightforward:

```python
@task(checkpoint=True, result=LocalResult(serializer=CustomSerializer()))
def read_data(filepath):
    # Do something with filepath
    return ...
```

How do I tell Prefect Orion that I want to use this serializer at the task level? The `task` decorator doesn't seem to support a custom serializer. 

```python
Signature:
task(
    __fn=None,
    *,
    name: str = None,
    description: str = None,
    tags: Iterable[str] = None,
    cache_key_fn: Callable[[ForwardRef('TaskRunContext'), Dict[str, Any]], Optional[str]] = None,
    cache_expiration: datetime.timedelta = None,
    retries: int = 0,
    retry_delay_seconds: Union[float, int] = 0,
)
```

Am I missing something? I couldn't find this information from the [docs](https://orion-docs.prefect.io/)

**Comment by zanieb** [comment]
Hi @andersy005, this is a great question!

We haven't exposed this yet. We are currently hard-coding the serializer to `cloudpickle`. For example:

Flows: https://github.com/PrefectHQ/prefect/blob/orion/src/prefect/engine.py#L397
Tasks: https://github.com/PrefectHQ/prefect/blob/orion/src/prefect/engine.py#L617-L619

You can see we've set up some tooling to make this easy to extend in the future, but we haven't decided what the best task/flow level API for configuring it is yet.

**Reply by andersy005** [reply]
Thank you for your prompt reply 🙏🏾, @madkinsz! I was starting to go down a rabbit hole in Prefect unit tests to see if there are any useful examples :). 

> You can see we've set up some tooling to make this easy to extend in the future, but we haven't decided what the best task/flow level API for configuring it is yet.

Orion looks great! Now that I know that I didn't miss anything, I'll just keep an eye out for future releases of Orion...
